### PR TITLE
Restructure pr curves notebook

### PR DESCRIPTION
**Pull request by roshankern**
This PR is ready for review!

In this PR, the class PR curves notebook is restructured to follow the structure of the evaluate notebooks.
This means all the models are evaluated on each data subset and the resulting data are saved in one compiled `.tsv. file.

### CODE REVIEW

File: 3.evaluate_model/scripts/nbconverted/class_PR_curves.py [hunk 1]
```diff
@@ -41,52 +42,62 @@
...
+        print(f"Evaluating {model_name} on dataset {label}")
+        
+        # load dataset (train, test, etc)
+        data = get_dataset(features_dataframe, data_split_indexes, label)
+        
+        fig, PR_data = class_PR_curves(data, model)
+        fig.suptitle(f"Precision Recall Curves for {model_name} on Dataset {label}")
+        plt.show()
+        
+        # add score data to compiled dataframe in tidy format
```
Comment (d33bs) [inline 3.evaluate_model/scripts/nbconverted/class_PR_curves.py]: I wasn't sure about what code this comment might reference (I may just misunderstand). Consider removing or moving it to a more appropriate location.

File: 3.evaluate_model/scripts/nbconverted/class_PR_curves.py [hunk 2]
```diff
@@ -41,52 +42,62 @@
...
+    # with nested for loops, we test each model on each dataset(corresponding to a label)
+    for label in data_split_indexes["label"].unique():
+        print(f"Evaluating {model_name} on dataset {label}")
+        
+        # load dataset (train, test, etc)
+        data = get_dataset(features_dataframe, data_split_indexes, label)
+        
+        fig, PR_data = class_PR_curves(data, model)
+        fig.suptitle(f"Precision Recall Curves for {model_name} on Dataset {label}")
+        plt.show()
```
Comment (d33bs) [inline 3.evaluate_model/scripts/nbconverted/class_PR_curves.py:72]: Really liked the dynamic titling here within the loop. Noticing the ability to control the figure and the similarity of output within the notebook, would it make sense to differentiate the line color (it's blue in all of them) to help the viewer distinguish each group?

**Review by d33bs — APPROVED** [review]
Nice work! I left a few suggestions but didn't see anything that would prevent a merge as-is. 

One follow up I wasn't sure about:
- I noticed your mention about using one `.tsv` but saw there were `.png` images which were also removed. Should these images be replaced by anything new?

File: 3.evaluate_model/scripts/nbconverted/class_PR_curves.py (hunk 2 above)
Comment (roshankern) [reply 3.evaluate_model/scripts/nbconverted/class_PR_curves.py:72]: I don't think that makes sense here as there are too many classes (16) to come up with a good color split. However, in the future when we do a mixed-multiclass model (4 classes) I will definitely implement specific colors.

File: 3.evaluate_model/scripts/nbconverted/class_PR_curves.py (hunk 1 above)
Comment (roshankern) [reply 3.evaluate_model/scripts/nbconverted/class_PR_curves.py]: Done in https://github.com/WayScience/phenotypic_profiling_model/pull/15/commits/8a4ddf864dd1b3922147b3770cd5d159e9c5afef

**Comment by roshankern** [comment]
Thanks for the review! These images should not be replaced by anything new. The format for the `3.evaluate_model` module is to save the intermediate data for evaluations in `.tsv` and keep the image evaluations in the notebooks.

**Pull request merged**